includes/orders/classes/class-order.php

```diff
@@ -384,8 +384,9 @@ public function get_adjustments() {
 	public function get_discounts() {
 		$discounts = array();
 
-		if ( empty( $this->adjustments ) ) {
-			return $discounts;
+		// Ensure adjustments exist.
+		if ( null === $this->adjustments ) {
+			$this->adjustments = $this->get_adjustments();
 		}
 
 		foreach ( $this->adjustments as $adjustment ) {
```

Comment:
This could just be refactored to do:

```suggestion
		foreach ( $this->get_adjustments() as $adjustment ) {
```

And remove the above `null` check. Might be a bit less repetitive.